torch/cuda/_utils.py

```diff
@@ -294,8 +294,10 @@ def __call__(
 
         for arg in args:
             if isinstance(arg, torch.Tensor):
-                if not arg.is_cuda:
-                    raise ValueError("All tensor arguments must be CUDA tensors")
+                if not arg.is_cuda and not (arg.is_cpu and arg.is_pinned()):
```

Comment:
@msaroufim please take a look at the changes in this file and let me know if you have any concerns. 

The corresponding commit is: https://github.com/pytorch/pytorch/pull/155372/commits/2f4480fe4249430a86018c39497941f7cca892a6